src/d/actor/d_a_obj_ebomzo.cpp

```diff
@@ -4,118 +4,299 @@
-    /* Nonmatching */
+    //Nonmatching
+
+    fopAc_ac_c* acActor;
+    cXyz diffVec;
+    u8 b, g, r;
+
+    if (mCollider.ChkTgHit()) {
+        acActor = mCollider.GetTgHitAc();
+        if (acActor) {
+            diffVec = mCollider.GetC() - acActor->current.pos;
+            if (diffVec.abs() < 100.0f) {
+                mXRotRate = 8;
+                current.angle.x += mXRotRate;
+
+                fopAcM_seStartCurrent(this, 0x69ac, 0);
+                mMode = Ebomzo_Mode_Demo;
+
+                fopAcM_onSwitch(this, prm_get_swSave());
+                
+                if (!mpParticleEmitter) {
+                    //color = tevStr.mColorK0;
+                    r = tevStr.mColorK0.r;
+                    g = tevStr.mColorK0.g;
+                    b = tevStr.mColorK0.b;
+                    mpParticleEmitter = dComIfGp_particle_set(0x828e, &current.pos, &current.angle);
+                    if (mpParticleEmitter) mpParticleEmitter->setGlobalPrmColor(r, g, b);
+                }
+            }
+        }
+    }
 }
 
 /* 00000864-000009BC       .text demo__Q211daObjEbomzo5Act_cFv */
 void daObjEbomzo::Act_c::demo() {
-    /* Nonmatching */
+    mXRotRate += mXRotRate / 8;
+    current.angle.x += mXRotRate;
+
+    if (current.angle.x >= 0x4000) {
+        current.angle.x = 0x4000;
+        mMode = Ebomzo_Mode_Fall;
+
+        dComIfGp_getVibration().StartShock(4, -33, cXyz(0.0, 1.0, 0.0));
+        fopAcM_seStartCurrent(this, 0x69ad, 0);
+
+        if (mpParticleEmitter) {
+            mpParticleEmitter->becomeInvalidEmitter();
+            mpParticleEmitter = NULL;
+        }
+
+        dComIfGp_particle_set(0x828f, &current.pos, &current.angle);
+    }
+
 }
 
 /* 000009BC-000009C0       .text fall__Q211daObjEbomzo5Act_cFv */
 void daObjEbomzo::Act_c::fall() {
-    /* Nonmatching */
+    return;
 }
 
 /* 000009C0-00000AFC       .text Execute__Q211daObjEbomzo5Act_cFPPA3_A4_f */
-void daObjEbomzo::Act_c::Execute(float(**)[3][4]) {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Execute(Mtx** matrix) {
+    cXyz offset;
+
+    offset.x = cM_ssin(current.angle.y) * 50.0f;
+    offset.y = 320.0;
+    offset.z = cM_scos(current.angle.y) * 50.0f;
+
+    mCollider.SetC(current.pos + offset);
+    dComIfG_Ccsp()->Set(&mCollider);
+
+    switch (this->mMode) {
+    case Ebomzo_Mode_Check: check(); break;
+    case Ebomzo_Mode_Demo: demo(); break;
+    case Ebomzo_Mode_Fall: fall(); break;
+    }
+
+    shape_angle = current.angle;
+    set_mtx();
+    *matrix = &M_tmp_mtx;
+
+    return TRUE;
 }
 
 /* 00000AFC-00000B9C       .text Draw__Q211daObjEbomzo5Act_cFv */
-BOOL daObjEbomzo::Act_c::Draw() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Draw() {
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, &tevStr);
+    g_env_light.setLightTevColorType(mpModel, &tevStr);
+    dComIfGd_setListBG();
+    mDoExt_modelUpdateDL(mpModel);
+    dComIfGd_setList();
+    dComIfG_Bgsp(); //In debug map but not used for anything??
```

Comment:
the debug map suggests `dComIfG_Bgsp()->DebugDrawPoly` was called here, which was probably a debug visualization function that got ifdef'd out for the retail version.

Reply:
Should I remove it then? For now, I just left it alone and removed the comment.